packages/web/src/lib/web-mode.ts

```diff
@@ -0,0 +1,133 @@
+import algosdk, { ConfirmedTxInfo, SuggestedParams, Transaction } from "algosdk";
+
+import { AlgoSigner, AlgoSignerSendTx, AlgoSignerSignedTx, AlgoSignerToBeSignedTx, ExecParams, TxParams } from "../types";
+import { mkTransaction } from "./txn";
+
+const CONFIRMED_ROUND = "confirmed-round";
+const LAST_ROUND = "last-round";
+
+export class WebMode {
+  algoSigner: AlgoSigner;
+  chainName: string;
+
+  constructor (algoSigner: AlgoSigner, chainName: string) {
+    this.algoSigner = algoSigner;
+    this.chainName = chainName;
+  }
+
+  /**
+   * wait for confirmation for transaction using transaction id
+   * @param txId Transaction id
+   */
+  async waitForConfirmation (txId: string): Promise<ConfirmedTxInfo> {
+    const response = await this.algoSigner.algod({
+      ledger: this.chainName,
+      path: '/v2/status'
+    });
+    console.log(response);
+    let lastround = response[LAST_ROUND];
+    while (true) {
+      const pendingInfo = await this.algoSigner.algod({
+        ledger: this.chainName,
+        path: `/v2/transactions/pending/${txId}`
+      });
+      if (
+        pendingInfo[CONFIRMED_ROUND] !== null &&
+        pendingInfo[CONFIRMED_ROUND] > 0
+      ) {
+        return pendingInfo;
+      }
+      lastround++;
+      await this.algoSigner.algod({
+        ledger: this.chainName,
+        path: `/v2/status/wait-for-block-after/${lastround}`// eslint-disable-line @typescript-eslint/restrict-template-expressions
+      });
+    }
+  }
+
+  /**
+   * Sender transaction to network
+   * @param signedTxn signed transaction
+   */
+  async sendTransaction (signedTxn: any): Promise<AlgoSignerSendTx> {
```

Comment:
why `any` ?

Reply:
`Algosigner` types will be used in another task, this PoC PR>.